mock-etcd/src/mock_etcd.rs

```diff
@@ -1161,38 +1270,39 @@ mod test {
         log::debug!("start e2e_lock_lease_test");
         smol::future::block_on(async {
             let lock_key012 = vec![0_u8, 1_u8, 2_u8];
+            let res = client
+                .lease()
+                .grant(EtcdLeaseGrantRequest::new(std::time::Duration::from_secs(
+                    10,
+                )))
+                .await
+                .unwrap_or_else(|err| panic!("failed to get key 0, the error is {}", err));
+            let lease_id = res.id();
 
-            let mut res = client
+            let _res = client
                 .lock()
-                .lock(EtcdLockRequest::new(lock_key012.clone(), 10))
+                .lock(EtcdLockRequest::new(lock_key012.clone(), lease_id))
                 .await
                 .unwrap_or_else(|err| panic!("failed to lock key012, the error is {}", err));
-            assert_eq!(res.take_key(), lock_key012);
+            // TODO: result key is not equal to lock_key012 when using real etcd
```

Comment:
It seems like you haven't fixed the bug related to the lock ye.

Reply:
Fixed.